Support extended PAN ID (optional) & secret network_key

felixstorm opened the pull request:
Support both extended PAN ID and network key *as strings*. That way
* extended PAN ID can be made optional (arrays cannot be made optional in the Hass.io add-on schema, so adding it - being mandatory - would break existing installations)
* the network key can be read from the secrets file (strings can be read from the secrets file whereas arrays can not)

Follow up to https://github.com/danielwelch/hassio-zigbee2mqtt/pull/263 and https://github.com/danielwelch/hassio-zigbee2mqtt/issues/259.

Example configuration:
```javascript
{
  // ...
  "advanced": {
    // ...
    "ext_pan_id_string": "1, 2, 3, 4, 5, 6, 7, 8",
    // ...
    "network_key": [],
    "network_key_string": "1, 3, 5, 7, 9, 11, 13, 15, 0, 2, 4, 6, 8, 10, 12, 13",
    // ...
  },
  // ...
}
```
Please note that `network_key` still needs to be present in the configuration (arrays being mandatory), but it may just be an empty array (it will be overwritten by the contents of `network_key_string` anyway).

felixstorm commented:
@Koenkk, @danielwelch Do you have any further suggestions or should I change anything here?

After Daniel's comments https://github.com/danielwelch/hassio-zigbee2mqtt/pull/263#issuecomment-561850043 and https://github.com/danielwelch/hassio-zigbee2mqtt/pull/263#issuecomment-562152426 I had tried to come up with a way to solve this that is completely backwards compatible - so please also let me know if there are any general concerns with this new solution here.

Or does any other workaround exist to use a custom extended PAN ID with the HA add-on that I might have overlooked?

Thanks and kind regards,  
Felix

danielwelch commented:
@felixstorm Sorry for the delay in response. We're just going to have to merge it and bite the bullet on the breaking change. I like to do that when there is a version bump, but other people have been merging those version updates lately without incorporating the config changes or updating the Changelog (which is totally fine, it just leads to things like this being pushed back in anticipation of the next version, then forgot about). I'll probably go ahead and merge a few of these sort of changes soon, hopefully this weekend.

felixstorm commented:
@danielwelch Thanks for answering and no worries about breaking changes with this PR. Since both the extended pan id and the secret network key have now been implemented as _additional & optional_ string settings, there should be no configuration change necessary for existing installations.

danielwelch commented:
Nice work, sorry, had been too long and didn’t notice which pull request I was responding to. Could we add documentation for the new option to the README? I’d like to have any options that “stray” from the standard zigbee2mqtt config options, with a rationale.

felixstorm commented:
Thanks! Happy to document the options in the README, but will probably only be able to do it within the next few days. Will add a commit here and let you know.

felixstorm commented:
@danielwelch I have updated the docs and rebased the PR to the current master branch. I have primarily updated the README inside the `zigbee2mqtt` folder since this is the one that is shown on the HA add-on page where the user can modify the configuration.

I also made three minor updates to the main README (one typo, one wording update to match the current HA version and IMHO there is no real need anymore to specify the MQTT server's port now) - feel free to change as you like.

Thanks,
Felix
